### PR TITLE
Autogenerate changelog generation

### PR DESCRIPTION
**Pull request by BBBmau**
### Description

Add ability to autogenerate changelog from command line by running
```
make changelog
make changelog-entry
```

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-helm/blob/main/CHANGELOG.md):

```release-note
NONE
```
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: .changelog/note.tmpl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{{- define "note" -}}
+{{.Body}}{{if not (stringHasPrefix .Issue "_")}} [[GH-{{- .Issue -}}](https://github.com/hashicorp/terraform-provider-helm/issues/{{- .Issue -}})]{{end}}
```
Comment (arybolovlev) [inline .changelog/note.tmpl:2]: @BBBmau, thanks for making this change. I think there is a typo here and the actual URL should contain `terraform-provider-kubernetes`.

File: CHANGELOG_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
+# How To Use
+
+Helm Provider uses `go-changelog` to generate its changelog on release:
```
Comment (arybolovlev) [inline CHANGELOG_GUIDE.md:3]: Kubernetes provider. 😊

**Review by arybolovlev — COMMENTED** [review]
Hi @BBBmau, thanks for making this change. I have left two small comments to address. 👍🏻

**Review by BBBmau — COMMENTED** [review]
@arybolovlev Updated!

**Pull request merged**